AnkiDroid/src/main/java/com/ichi2/utils/ClipboardUtil.kt

```diff
@@ -68,42 +69,39 @@ object ClipboardUtil {
 }
 
 /**
- * Copies the provided text to the clipboard (truncated if necessary)
- * @return `true` if the clipboard is modified. `false` otherwise
+ * Copies the provided [text] to the clipboard (truncated if necessary),
+ * and show either a snackbar, if possible, or a toast in case of API>32,
+ * with the success or failure message.
  */
-fun Context.copyToClipboard(text: String): Boolean {
-    val clipboardManager = this.getSystemService(Activity.CLIPBOARD_SERVICE) as? ClipboardManager
-    if (clipboardManager == null) {
-        Timber.w("Failed to obtain ClipboardManager")
-        return false
-    }
-
-    clipboardManager.setPrimaryClip(
-        ClipData.newPlainText(
-            "${VersionUtils.appName} v${VersionUtils.pkgVersionName}",
-            text
-        )
-    )
-    return true
-}
-
-/**
- * Copy given [text] to the clipboard,
- * and show either a snackbar, if possible, or a toast, with the success or failure message.
- *
- * @see copyToClipboard
- */
-fun Context.copyToClipboardAndShowConfirmation(
+fun Context.copyToClipboard(
     text: String,
     @StringRes successMessageId: Int,
     @StringRes failureMessageId: Int
 ) {
-    val successfullyCopied = copyToClipboard(text)
+    var copied = false
+    val clipboardManager = this.getSystemService(Activity.CLIPBOARD_SERVICE) as? ClipboardManager
+    if (clipboardManager == null) {
+        Timber.w("Failed to obtain ClipboardManager")
+    } else {
+        copied = try {
+            clipboardManager.setPrimaryClip(
+                ClipData.newPlainText(
+                    "${VersionUtils.appName} v${VersionUtils.pkgVersionName}",
+                    text
+                )
+            )
+            true
+        } catch (e: Exception) {
+            Timber.e("Failed to copy text to clipboard", e)
```

Comment:
The `e` parameter comes first in Timber

Use `Timber.w` for non-fatal errors, not `Timber.e`


```suggestion
            Timber.w(e, "Failed to copy text to clipboard")
```